src/main/cpp/src/hive_hash.cu

```diff
@@ -197,8 +210,16 @@ class hive_device_row_hasher {
    public:
     using hash_functor_t = cudf::experimental::row::hash::element_hasher<hash_function, Nullate>;
 
-    __device__ element_hasher_adapter(Nullate check_nulls) noexcept
-      : hash_functor{check_nulls, HIVE_INIT_HASH, HIVE_INIT_HASH}
+    __device__ element_hasher_adapter(Nullate check_nulls,
+                                      cudf::size_type col_idx,
```

Comment:
`col_idx` is actually the flattened index, so `flattened_idx` seems to be a more expressive name.